### PR TITLE
fix kpng version command and standardize build command

### LINKED ISSUE #228: kpng version shows unknown
# Issue
`kpng version` should show the git tag/hash id which the build was based or the release version.

Reference:
[1] https://github.com/kubernetes-sigs/kpng/pull/219/commits/2a70684c55a5351d903d6b42d277378ac889c982#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R20

[2] https://github.com/kubernetes-sigs/kpng/pull/219/commits/2a70684c55a5351d903d6b42d277378ac889c982#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R21

### PR DESCRIPTION
**Pull request by jaehnri**
fixes #228 

The problem here likely stems from the fact that, currently, [there are too many ways to build and deploy KPNG](https://kubernetes.slack.com/archives/C02GA1RSCLR/p1673117536594379). None of them were passing the commit SHA to go build/install.

This PR:
- fixes the version command for `make tilt-up`, `make build`,  `kpng-local-up.sh`, and `test_e2e.sh`;
- makes all the tools above build `kpng` with the same command (`make image`);

### CODE REVIEW

File: cmd/kpng/version.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+		Run: func(cmd *cobra.Command, args []string) {
+			fmt.Println(kpngVersion())
+		},
+	}
+}
+
+// kpngVersion returns information about the version.
+func kpngVersion() string {
+	return fmt.Sprintf(`-------------------------------------------------------------------------------
+Kubernetes Proxy NG
```
Comment (jaehnri) [inline cmd/kpng/version.go:47]: BTW, what is the meaning of NG here?

Comment (mcluseau) [reply cmd/kpng/version.go:47]: next gen

**Comment by jaehnri** [comment]
Marked as draft because it is not solved for `test_e2e.sh` and `modd`. Will try to standardize the build process later.

**Comment by friedrichwilken** [comment]
@jaehnri can this be closed?

**Comment by jaehnri** [comment]
@friedrichwilken it looks like my last commit made the pipeline fail. I'll check it very soon so it can be reviewed.

File: cmd/kpng/main.go [hunk 2]
```diff
@@ -96,13 +94,3 @@ func setupGlobal() (ctx context.Context) {
 
 	return
 }
-
-func versionCmd() *cobra.Command {
-	return &cobra.Command{
-		Use:   "version",
-		Short: "print the version and quit",
-		Run: func(cmd *cobra.Command, args []string) {
-			fmt.Println(version)
```
Comment (jaehnri) [inline cmd/kpng/main.go:105]: This was the only reference to `fmt` in this file. However, for some reason, I couldn't remove its import. Even importing it with `_` for side effects did not work. Not sure why.

**Review by qabpea — APPROVED** [review]

**Review by qabpea — APPROVED** [review]

**Pull request merged**